# HG changeset patch
# User Rik <rik@octave.org>
# Date 1523915983 25200
#      Mon Apr 16 14:59:43 2018 -0700
# Node ID 592a4258b237da39902a9a2476ce76a0d3fd35dd
# Parent  308a272bbe3c61185d9920b34e510293dd9b4c6d
logspace.m: Update BIST tests to pass after cset ca022a8c4015.

* logspace.m: Use complex() function to create complex numbers with Inf parts.

diff --git a/scripts/general/logspace.m b/scripts/general/logspace.m
--- a/scripts/general/logspace.m
+++ b/scripts/general/logspace.m
@@ -95,18 +95,18 @@ endfunction
 %! x4 = logspace (1, pi, 10);
 %! assert (size (x1) == [1, 50] && abs (x1(1) - 10) < eps && abs (x1(50) - 100) < eps);
 %! assert (size (x2) == [1, 10] && abs (x2(1) - 10) < eps && abs (x2(10) - 100) < eps);
 %! assert (size (x3) == [1, 10] && abs (x3(1) - 10) < eps && abs (x3(10) - 0.01) < eps);
 %! assert (size (x4) == [1, 10] && abs (x4(1) - 10) < eps && abs (x4(10) - pi) < sqrt (eps));
 %!assert (logspace (Inf, Inf, 3), [Inf, Inf, Inf])
 %!assert (logspace (-Inf, -Inf, 3), [0, 0, 0])
 %!assert (logspace (-Inf, Inf, 3), [0, NaN, Inf])
-%!assert (logspace (Inf + 1i, Inf + 1i, 3), [-Inf + Inf * 1i, -Inf + Inf * 1i, -Inf + Inf * 1i])
-%!assert (logspace (-Inf + 1i, Inf + 1i, 3), [0, NaN + NaN * 1i, -Inf + Inf * 1i])
+%!assert (logspace (Inf + 1i, Inf + 1i, 3), repmat (complex (-Inf,Inf), [1, 3]))
+%!assert (logspace (-Inf + 1i, Inf + 1i, 3), [0, NaN + NaN * 1i, complex(-Inf, Inf)])
 %!assert (logspace (0, Inf, 3), [1, Inf, Inf])
 %!assert (logspace (0, -Inf, 3), [1, 0, 0])
 %!assert (logspace (-Inf, 0, 3), [0, NaN, 1])
 %!assert (logspace (Inf, 0, 3), [Inf, NaN, 1])
 %!assert (logspace (Inf, -Inf, 3), [Inf, NaN, 0])
 
 ## Test input validation
 %!error logspace ()
